server/services/interventionsService.test.ts

```diff
@@ -72,6 +72,36 @@ pactWith({ consumer: 'Interventions UI', provider: 'Interventions Service' }, pr
       })
     })
 
+    describe('for a referral that has had a service provider selected', () => {
+      beforeEach(async () => {
+        await provider.addInteraction({
+          state:
+            'There is an existing draft referral with ID of d496e4a7-7cc1-44ea-ba67-c295084f1962, and it has had a service provider selected',
```

Comment:
> and it has had a service provider selected

I'm still confused about when this selection happens: before draft referrals are created or during... Not a blocker, though!

Reply:
I think it's when we've gone through the "Find" process and an intervention has been selected, but I could be wrong about that...

Reply:
yeh the service itself (and the service user) will be selected right from the start.

Reply:
Yeah, good point. When we first started writing these contract tests, it was with the original user journey in mind, where the draft referral is created and then the user Find-s an intervention (i.e. service provider and service category). Now that Find is done before the user starts the referral form, we _could_ make the service category ID and service provider ID non-nullable on a draft referral, but I've kept it nullable to be consistent with [the corresponding test for `serviceCategoryId`](https://github.com/ministryofjustice/hmpps-interventions-ui/blob/672a957cf3aecd976639682c11de0c82ce2672a5/server/services/interventionsService.test.ts#L44-L73).